### ISSUE TITLE
Dotmarker pops on top of day when switching to another month when the standard selected day has a dot programmed on it

### ISSUE THREAD

**Issue by martjemeyer** [Mozharovsky/CVCalendar#59]
When I change from month A to month B, when the autoselected day in the next month has a dot programmed on it, the dot appear on top of that day.
So in the first screenshot, the date selected in the new month is 20, which has no dot programmed on it.
The other screen, the date jumps to the (in this case) today date, which has a dot on it and puts it on top of the day, almost in the box off the day above.
If I set the day to 28's from a previous month, the same effect is generated.

Also, when I return from any popup screen (where I might add something on that day) is there a way to refresh the dot function? So I don't have to scroll to another month and back.

Thanks!

![screenshot 2015-04-27 19 35 12](https://cloud.githubusercontent.com/assets/9798987/7353804/b2184506-ed16-11e4-9ecb-2e6a912d83ad.png)
![screenshot 2015-04-27 19 35 28](https://cloud.githubusercontent.com/assets/9798987/7353809/b6550438-ed16-11e4-8061-aff9d5ecfe09.png)

**Comment by martjemeyer** [comment]
Ah, I think its because of my implementation.
I set the date with toggleViewWithDate(date) in the ViewDidLoad, since I can set the date in another view .
So when loading the monthView, it gets todays date in the viewDidLoad or a preselected date. If that date appears to have a dot marker installed, the above behaviour appears.
I think that did not happen before. 

Also I just noticed, if I do the same for a complete different month than the current month. So if I navigate from my view which only displays one day at the time , I navigate to another month, lets say from may to june and jump to the monthview, the monthView is blank. So again in my viewDidLoad I set the current day with calendarView.toggleViewWithDate(date)..

Is there a better method I should use? So I need to be able to set a date when the monthView loads which I preselected in another view or when a popup closes..

The toggle month to week toggle function works great btw! Really nice job!

**Comment by Mozharovsky** [comment]
Hello @martjemeyer, 

Couldn't you please describe steps I should go i order to reproduce blank month view appearing? 

Thanks in advance, 
Eugene

**Comment by martjemeyer** [comment]
Hmm I have trouble to reproduce it on your demo app. I'll come back at this tomorrow if I know what happens different.

**Comment by martjemeyer** [comment]
I tried everything to reproduce or even deleting all code from the monthviewcontroller, to get to the point where it goes wrong. But no luck. I expect it goes somewhere wrong from my segue to the month viewcontroller and the data I submit. But even that I tried to add to the demo app but still no luck:)

In the end I solved the problem, by switching to the date I want in the viewDidApepar method. Then it also nicely animates. So this one can be closed, since I think doing it the way I do now is even more prettier;)

Thank you for you great work like always!

Regard,

Henk-Martijn
